lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -126,6 +126,20 @@ export class ResponseHandler {
                     throw ClientAuthError.createNonceMismatchError();
                 }
             }
+
+            // token max_age check
+            if (request.maxAge) {
+                const authTime = idTokenObj.claims.auth_time;
+
+                if (!authTime) {
+                    throw ClientAuthError.createAuthTimeNotFoundError();
+                }
+
+                const twoMinuteSkew = 120000; // two minutes in milliseconds
+                if ((Date.now() - twoMinuteSkew) > (authTime + request.maxAge)) {
+                    throw ClientAuthError.createMaxAgeTranspiredError();
```

Comment:
I wonder if reusing this error would be appropriate? https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dcaef482889239a4fc3da0f0dee56d1551a3ced8/lib/msal-common/src/error/ClientAuthError.ts#L488

Reply:
I see that directly above on line 126, a mismatched nonce has it's own error message. Therefore I believe these max_age errors should have their own error messages as well. Thoughts?

Reply:
I would argue that a token needing to be expired to due max age is potentially semantically the same as a natural token expiration error (i.e. they could be handled in the same way), and thus could have the same error (maybe with a different sub error code?).

Reply:
Ok, I can see changing the second error message to the existing tokenRefreshRequired one like you suggested. However, I think I should still have the authTimeNotFoundError message on line 135 in ResponseHandler.ts. Thoughts?

Reply:
Hmm maybe having a different error is fine, curious what others think. And yes, the second error can stay.

Reply:
Also, I don't think this is the right place for this logic. This code will only run after a network request, but we would want this to run after retrieving a cache token as well, correct? If so, it should potentially go here? https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dcaef482889239a4fc3da0f0dee56d1551a3ced8/lib/msal-browser/src/app/PublicClientApplication.ts#L195

Reply:
You're right - I forgot about the cache scenario. However the line you suggested still only runs in the case of a network call, inside of acquireSilentTokenAsync. I think the logic should be in two different places. I put it where it currently is in ResponseHandler to mimic the python functionality - I still think it should be there. But I will look into re-using the logic for the cache scenario.

Reply:
@Robbie-Microsoft if you renew a token silently using a refresh token, does the `auth_time` reset? If so, then the logic can be potentially move to here (where we have the existing logic for checking if a cached token has expired): https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dcaef482889239a4fc3da0f0dee56d1551a3ced8/lib/msal-common/src/client/SilentFlowClient.ts#L48-L95

Reply:
@Robbie-Microsoft That line runs for both cached and non-cached requests: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dcaef482889239a4fc3da0f0dee56d1551a3ced8/lib/msal-browser/src/app/PublicClientApplication.ts#L190

Reply:
I think we should leave the check where it is in ResponseHandler.ts, but also add it right after [SilentCacheClient.tsL26](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/src/interaction_client/SilentCacheClient.ts#L26). If we move it to where you suggested, then in the case of a network request when the check finally runs, the network request has already been completed. Or has it already been completed at the current point in ResponseHandler.ts anyways? Thoughts?

Reply:
If we take it out of where it's currently located in ResponseHandler.ts, then a CacheRecord will be generated (and other code will run too) even when there is a max_age error.

Reply:
I think thats fine, as long as you create a helper function so that the `auth_time` checking logic isn't duplicated.

Reply:
(as a function on the `AuthToken` class could be one place)